core/src/main/scripts/migrate_db.py

```diff
@@ -150,6 +154,32 @@ def is_version_larger(version1, version2):
         return True
     return False
 
+def check_reference_genome(portal_properties, cursor, force_migration):
+    """ Retrieve reference genomes from database """
+    try:
+        sql_statement = """
+                           select count(distinct NCBI_BUILD), CANCER_STUDY_IDENTIFIER
+                           from mutation_event 
+                           join mutation on mutation.MUTATION_EVENT_ID = mutation_event.MUTATION_EVENT_ID 
+                           join genetic_profile on genetic_profile.GENETIC_PROFILE_ID = mutation.GENETIC_PROFILE_ID
+                           join cancer_study on cancer_study.CANCER_STUDY_ID = genetic_profile.CANCER_STUDY_ID
+                           where NCBI_BUILD not in ('NA','')
+                           group by CANCER_STUDY_IDENTIFIER
+                       """
+        cursor.execute(sql_statement)
+        for row in cursor.fetchall():
+            ref_count, study = row
+            if int(ref_count) > 1 and not force_migration:
```

Comment:
This test is good for catching whether there are multiple reference genomes present in the mutation event table. But what if there is only a single value for genome build in the mutation event table ... but that value differs from one of our recognized patterns {'37', 'hg19', 'GRCh37', '38', 'hg38', 'GRCh38'}. Let's say everywhere in mutation_event records for a study, the ncbi_build string is "UNKNOWN". I think this could also  generate a warning here, rather than proceeding to import the study with no warning and associate it with genome build hg19 (the default) during migration. Maybe something like:
allowable_genome_references = set( ['37', 'HG19', 'GRCH37', '38', 'HG38', 'GRCH38'] )
if not (retrieved_ncbi_build.upper() in allowable_genome_references):
give warning and exit

this would require changing the query to something like select group_concat(ncbi_build), count(distinct ncbi_build), .... and the later:
retrieved_ncb_build, ref_count, study = row

Maybe this goes beyond what we originally asked for though ... I guess the whole point of having the migration script default to hg37 is that you don't need to check for every possible "other" valid genome specification. So if it is ok that "UNKNOWN" will be silently migrated to hg19, then maybe this comment can be ignored.

Reply:
@sheridancbio , good point! will add logic in. Thanks!

Reply:
@sheridancbio , I've made changes as you suggested and checked in the code. Thanks!